modules/workers/variables.tf

```diff
@@ -48,6 +48,22 @@ variable "kubeproxy_mode" { type = string }
 variable "max_pods_per_node" { type = number }
 variable "node_labels" { type = map(string) }
 variable "node_metadata" { type = map(string) }
+variable "platform_config" {
+  default = null
+  type = object({
+    type                                           = string,
+    are_virtual_instructions_enabled               = optional(bool),
```

Comment:
I see you switched to optional map params, nice!

If you use something like `are_virtual_instructions_enabled = optional(bool, null)`, you can get rid of the lookup in the maps above, as the value will always be present in the map, if the user doesn't give it it's the default, `null`.
It looks way less messy at the call site, but I understand it can be a stylistic choice.

See: https://developer.hashicorp.com/terraform/language/expressions/type-constraints#optional-object-type-attributes

Reply:
Thanks - I did end up taking advantage of this at your suggestion in the other PR for `var.nsgs` and `var.subnets`.

On further review of this case, the lookups are made against a local map rather than the input variable to honor pool-level overrides merged onto global defaults if specified - though I think it could become cleaner with `worker_pools` defined as `map(object)` in the future.

Reply:
Oh, I'm sorry, it's hard to tell from GitHub, but I'm happy you found the info useful.

I find that this level of static typing also alleviates much of the need for documentation, so hopefully you guys get less annoyed by people in the future.